### PR TITLE
New .NET Generator with new expected files

### PR DESCRIPTION
**Pull request by assyadh**
Hi,

This contains a new generator for .NET written in typescript with the new expected files
This also removes of the old generator codebase.

Fixes https://github.com/awslabs/jsii/issues/463
Fixes https://github.com/awslabs/jsii/issues/528

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by RomainMuller** [comment]
This is currently broken due to lingering references to the old packages here and there.

**Comment by assyadh** [comment]
> This is currently broken due to lingering references to the old packages here and there.

Got it. Was missing a packages.json update. it builds now.

File: packages/jsii-pacmak/test/expected.jsii-calc/dotnet/Amazon.JSII.Tests.CalculatorPackageId/Amazon/JSII/Tests/CalculatorNamespace/ClassThatImplementsTheInternalInterface.cs [hunk 1]
```diff
@@ -2,7 +2,9 @@
 
 namespace Amazon.JSII.Tests.CalculatorNamespace
 {
-    /// <remarks>stability: Experimental</remarks>
+    /// <remarks>
+    /// stability: Experimental
+    /// </remarks>
     [JsiiClass(nativeType: typeof(ClassThatImplementsTheInternalInterface), fullyQualifiedName: "jsii-calc.ClassThatImplementsTheInternalInterface")]
     public class ClassThatImplementsTheInternalInterface : DeputyBase, IINonInternalInterface
```
Comment (costleya) [inline packages/jsii-pacmak/test/expected.jsii-calc/dotnet/Amazon.JSII.Tests.CalculatorPackageId/Amazon/JSII/Tests/CalculatorNamespace/ClassThatImplementsTheInternalInterface.cs]: I see we are implicitly implementing the interfaces. What happens when we implement 2 interfaces with the same method signatures? Can't we always explicitly implement?

File: packages/jsii-pacmak/lib/targets/dotnet/dotnetgenerator.ts [hunk 2]
```diff
@@ -272,9 +272,16 @@ export class DotNetGenerator extends Generator {
             this.code.line();
         }
 
+        if (initializer) {
```
Comment (costleya) [inline packages/jsii-pacmak/lib/targets/dotnet/dotnetgenerator.ts]: would making the null check in the "emitIfNecessary" be cleaner?

File: packages/jsii-pacmak/lib/targets/dotnet/dotnetgenerator.ts [hunk 3]
```diff
@@ -272,9 +272,16 @@ export class DotNetGenerator extends Generator {
...
+            this.dotnetRuntimeGenerator.emitDeprecatedAttributeIfNecessary(initializer);
+        }
         this.code.openBlock(`protected ${className}(ByRefValue reference): base(reference)`);
```
Comment (costleya) [inline packages/jsii-pacmak/lib/targets/dotnet/dotnetgenerator.ts]: Can't properties and methods also be "deprecated" instead of just at the class level?

Comment (assyadh) [reply packages/jsii-pacmak/lib/targets/dotnet/dotnetgenerator.ts]: It's handled when emitting the attributes.

Here is an example for methods: https://github.com/awslabs/jsii/blob/677645dc734dc5f2abb059c05eb295aa939524eb/packages/jsii-pacmak/lib/targets/dotnet/dotnetruntimegenerator.ts#L74

File: packages/jsii-pacmak/test/expected.jsii-calc/dotnet/Amazon.JSII.Tests.CalculatorPackageId/Amazon/JSII/Tests/CalculatorNamespace/ClassThatImplementsTheInternalInterface.cs (hunk 1 above)
Comment (assyadh) [reply packages/jsii-pacmak/test/expected.jsii-calc/dotnet/Amazon.JSII.Tests.CalculatorPackageId/Amazon/JSII/Tests/CalculatorNamespace/ClassThatImplementsTheInternalInterface.cs]: You mean on the inherited/implemented classe/interfaces? 

Sure, I can make a change to just always put the fully qualified type.

File: packages/jsii-pacmak/lib/targets/dotnet/dotnetgenerator.ts (hunk 2 above)
Comment (assyadh) [reply packages/jsii-pacmak/lib/targets/dotnet/dotnetgenerator.ts]: Sure. will move the check around.

File: packages/jsii-pacmak/test/expected.jsii-calc/dotnet/Amazon.JSII.Tests.CalculatorPackageId/Amazon/JSII/Tests/CalculatorNamespace/ClassThatImplementsTheInternalInterface.cs (hunk 1 above)
Comment (costleya) [reply packages/jsii-pacmak/test/expected.jsii-calc/dotnet/Amazon.JSII.Tests.CalculatorPackageId/Amazon/JSII/Tests/CalculatorNamespace/ClassThatImplementsTheInternalInterface.cs]: Just for reference: https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/interfaces/explicit-interface-implementation

Comment (assyadh) [reply packages/jsii-pacmak/test/expected.jsii-calc/dotnet/Amazon.JSII.Tests.CalculatorPackageId/Amazon/JSII/Tests/CalculatorNamespace/ClassThatImplementsTheInternalInterface.cs]: Actually, it looks like the work may be more involved, I don't want to delay this PR (Which the goal is to be a 1:1 against the OLD generator) so I created this issue to address the problem:

https://github.com/awslabs/jsii/issues/560

**Pull request merged**